packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts

```diff
@@ -78,7 +78,54 @@ export class PayPalNotificationHandler extends PayPalHandler {
     }
   }
 
-  private async handleMpCancel(message: IpnMerchPmtType) {}
+  /**
+   * Handle merchant payment cancelled notification from PayPal
+   * and update PayPalBillingAgreements to "Cancelled" status
+   * and removes the billing agreement from Stripe Customer
+   *
+   * @param message
+   */
+  private async handleMpCancel(message: IpnMerchPmtType) {
+    const billingAgreement = await getPayPalBAByBAId(message.mp_id);
+    if (!billingAgreement) {
+      this.log.error('handleMpCancel', {
+        message: 'Billing agreement not found',
+        ipnMessage: message,
+      });
+      return;
+    }
+    if (billingAgreement.status == 'Cancelled') {
+      return;
+    }
+    const accountCustomer = await getAccountCustomerByUid(billingAgreement.uid);
+    const customer = await this.stripeHelper.fetchCustomer(
```

Comment:
I see you couldn't use `customer` because it needs the email address. Unfortunately this function does not auto-expand anything, so the subscriptions are not guaranteed to be expanded. The additional expand argument should be provided here:
```
      'sources',
      'subscriptions',
      'invoice_settings.default_payment_method',
```